sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/cli/XrayWrapperArtifactoryClientSupport.java

```diff
@@ -0,0 +1,237 @@
+        while (!scanned) {
+            scanned = handleScanStatus(context);
+        }
+
+        // save reports from xray
+        if (artifactoryClient.requestScanReports()) {
+            // rewrite cycloneDX report with security report information
+            readAndConvertReports();
+        }
+
+        // delete artifact from artifactory
+        artifactoryClient.deleteArtifact();
+        artifactoryClient.deleteUploads();
+    }
+
+    private void readAndConvertReports() throws XrayWrapperException {
+        XrayWrapperReportData xrayWrapperReportData = new XrayWrapperReportData();
+
+        XrayWrapperReportSupport.XrayReportFiles reportFiles = reportReader.collectXrayReportsInArchive(xrayWrapperConfiguration.getZipDirectory(),
+                xrayWrapperConfiguration.getXrayPdsReport());
+        xrayWrapperReportData.setSecurityReport(reportFiles.securityReport());
+        xrayWrapperReportData.setCycloneReport(reportFiles.cycloneReport());
+        xrayWrapperReportData.setXrayPdsReport(reportFiles.xrayPdsReport());
+
+        xrayWrapperReportData.setCycloneDXVulnerabilityHashMap(reportReader.readSecurityReport(xrayWrapperReportData.getSecurityReport()));
+        xrayWrapperReportData
+                .setSbom(reportReader.mapVulnerabilities(xrayWrapperReportData.getCycloneReport(), xrayWrapperReportData.cycloneDXVulnerabilityHashMap));
+        reportReader.writeReport(xrayWrapperReportData.getSbom(), xrayWrapperReportData.getXrayPdsReport());
+    }
+
+    /**
+     * checks on the scan status of the artifact in periodically
+     *
+     * @return true if the artifact is scanned
+     * @throws XrayWrapperException
+     */
+    private boolean handleScanStatus(ClientSupportContext context) throws XrayWrapperException {
+        ScanStatus status = artifactoryClient.getScanStatus();
+        LOG.debug("Artifact status is: " + status.getStatusValue());
+        if (context.isTimeoutReached()) {
+            throw new XrayWrapperException("Reached maximum scan timeout of " + xrayWrapperConfiguration.getMaxScanDurationHours() + " hours. "
+                    + "Started scan at: " + context.startTime, XrayWrapperExitCode.TIMEOUT_REACHED);
+        }
+        switch (status) {
+        case NOT_SCANNED -> {
+            waitSeconds(xrayWrapperConfiguration.getWaitUntilRetrySeconds());
+            if (context.isMaximumRetriesReached()) {
+                context.resetRetries();
+                startNewScan();
+            }
+            context.markNextRetry();
+            return false;
+        }
+        case IN_PROGRESS -> {
+            waitSeconds(xrayWrapperConfiguration.getWaitUntilRetrySeconds());
+            return false;
+        }
+        case SCANNED -> {
+            return true;
+        }
+        }
+        throw new XrayWrapperException("Received unexpected scan status", XrayWrapperExitCode.UNSUPPORTED_API_REQUEST);
+    }
+
+    private static void waitSeconds(int val) throws XrayWrapperException {
```

Comment:
I think we can remove the static part here - its is only private and used internally, so access should be always possible.
```suggestion
    private void waitSeconds(int val) throws XrayWrapperException {
```